### PR TITLE
[POAE7-2935] copy elimination for some bare columns

### PR DESCRIPTION
**Pull request by Ma-Jian1**
### What changes were proposed in this pull request?
copy elimination for some bare columns

In `select a, b, b+c from test`, column `a` is not toughed in the project, column `b` is not modified,
these columns should not allocate new memory and copy from input to output.
This pr we skip columns `a` and `b` in the project and just copy the children pointer in `resetBatch` and clear
input children pointer in the `LazyNode` to ensure these columns are not released.

Something needs to be improved:
1. maybe we can remove `LazyNode`, and just do the work in the deferred function, then the StageNode can be removed.
2. we just support copy elimination in the project, skipped if has others op(eg. filter, join)
3. `resetBatch` should match input reuse(ie. if we reuse the input, resetBatch must be passed the input array and schema)
4. project op should be removed in `select a, b from test`, just keep the lazy op

### Why are the changes needed?
performance optimization

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT and benchmark

### Which label does this PR belong to?
INFRA

### CODE REVIEW

File: cpp/src/cider-velox/benchmark/expression/PipelineOperator.cpp [hunk 1]
```diff
@@ -293,11 +296,16 @@ BENCHMARK_RELATIVE(nextgenOpt) {
...
+  // warmup: preload YAML
+  FunctionLookupEngine::getInstance(PlatformType::PrestoPlatform);
+
   benchmark = std::make_unique<PipelineOperator>();
   // mimic tpch q1 project expression
   benchmark->initialize(
       {"c", "c_1", "d", "d_1", "d_2", "d_3"},
       {"c", "c_1", "d", "d_1", "d_1*(1.0-d_2)", "d_1*(1.0-d_2)*(1.0+d_3)", "d_2"});
+  // {"i64", "i64_1", "d", "d_1", "d_2", "d_3"},
+  // {"i64", "i64_1", "d", "d_1", "d_1*(1.0-d_2)", "d_1*(1.0-d_2)*(1.0+d_3)", "d_2"});
```
Comment (bigPYJ1151) [inline cpp/src/cider-velox/benchmark/expression/PipelineOperator.cpp:308]: Such code could be removed if unused.

File: cpp/src/cider-velox/benchmark/tpch/TpchInMemBuilder.cpp [hunk 2]
```diff
@@ -151,7 +151,8 @@ TpchPlan TpchInMemBuilder::getQ1Plan(double scaleFactor) const {
           //   .tableScan(kLineitem, selectedRowType, fileColumnNames, {filter})
           .tableScan(Table::TBL_LINEITEM, std::move(selectedColumns), scaleFactor)
           .capturePlanNodeId(lineitemPlanNodeId)
-          .filter(filter)
+          // velox filter will wrap result childrens in dictionary
+          //   .filter(filter)
```
Comment (bigPYJ1151) [inline cpp/src/cider-velox/benchmark/tpch/TpchInMemBuilder.cpp:155]: Comment filters for debugging? Better to revert it if so.

File: cpp/src/cider/exec/nextgen/operators/OpNode.h [hunk 3]
```diff
@@ -57,7 +56,7 @@ class OpNode : public std::enable_shared_from_this<OpNode> {
   /// \brief The name of the operator node
   const char* name() const { return name_; }
 
-  void setInputOpNode(const OpNodePtr& input) { input_ = input; }
+  // void setInputOpNode(const OpNodePtr& input) { input_ = input; }
```
Comment (jikunshang) [inline cpp/src/cider/exec/nextgen/operators/OpNode.h:59]: remove?

**Review by jikunshang — APPROVED** [review]

File: cpp/src/cider-velox/benchmark/tpch/TpchInMemBuilder.cpp (hunk 2 above)
Comment (Ma-Jian1) [reply cpp/src/cider-velox/benchmark/tpch/TpchInMemBuilder.cpp:155]: copy elimination doesn't support filter. So I commend all filter operations and just benchmark project op.

**Review by jikunshang — APPROVED** [review]

**Review by bigPYJ1151 — APPROVED** [review]

**Pull request merged**